v2/devdoc/log_context_property_type_ascii_char_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)(const char* format, ...);
+
+extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr);
+
+```
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` produces the string representation of an `ascii_char_ptr`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_017: [** If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_002: [** If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall return the length of the string pointed to by `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_003: [** Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall copy the string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%s` and pass in the values list the `const char*` value pointed to be `property_value`. **]**
+
+Note: the copying of the string can be somewhat optimized by not using `snprintf`, this will be considered in a separate task.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_004: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_005: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` copies the `ascii_char_ptr` value from the address pointed by `src_value` to `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_006: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_007: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall copy the entire string (including the `NULL` terminator) from `src_value` to `dst_value`. **]**
```

Comment:
>the entire string [](http://example.com/codeflow?start=127&length=17)

Since this functon doesn't allocate memory for `dst_value` (and there's no `size_t dst_value_size`) what happens when `src_value`'s length is greater than `dst_value`?

Or that "cannot happen" by design? If that's the case, it should be spelled out explicitly.

Reply:
Let's say it is by design. I added a sentence about it.